### PR TITLE
[Feature] Make sentry error less annoying [OSF-7823]

### PR DESCRIPTION
**Pull request by fabmiz**
## Purpose

Make sentry aggregate errors 

## Changes

- Modify sentry (flask client)  config to take extra data as optional keyword arg
- Modify website view to call sentry with guid as extra data

## Ticket

https://openscience.atlassian.net/browse/OSF-7823

### CODE REVIEW

File: framework/sentry/__init__.py [hunk 1]
```diff
@@ -35,13 +35,16 @@ def log_exception():
     })
 
 
-def log_message(message):
+def log_message(message, extra_data={}):
```
Comment (sloria) [inline framework/sentry/__init__.py]: Drive-by comment: Don't use mutable default arguments: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

File: website/views.py [hunk 2]
```diff
@@ -246,7 +246,7 @@ def resolve_guid(guid, suffix=None):
         #   expected.
         if not hasattr(guid_object.referent, 'deep_url'):
             sentry.log_message(
-                'Guid `{}` resolved to an object with no deep_url'.format(guid)
+                'Guid resolved to an object with no deep_url', dict(guid = guid)
```
Comment (sloria) [inline website/views.py]: Drive-by: remove extra spacing around `=`, or do `{'guid': guid}`.

File: framework/sentry/__init__.py (hunk 1 above)
Comment (fabmiz) [inline framework/sentry/__init__.py]: Thanks a lot for the tip!
